intermediate-week-1/middlewares/auth.js

```diff
@@ -1,9 +1,28 @@
+const User = require('../models/User.js');
+const { verifyAccessToken } = require('../utils/tokenUtils.js');
+
 exports.requireSignIn = async (req, res, next) => {
-  const authorization = req.headers.authorization;
-  const [bearer, token] = authorization.split(' ');
+  try {
+    const authorization = req.headers.authorization;
+    const [bearer, token] = authorization.split(' ');
+
+    /**
+     * verify the token and verify if user is logged in in or not
+     * If user is logged in then call the next() function to go to the next middleware
+     */
+    if (token && bearer === 'Bearer') {
+      const { payload } = verifyAccessToken(token);
+      if (!payload) throw new Error('Invalid Token');
+
+      req.user = await User.findById({ _id: payload._id });
+      return next();
```

Comment:
Nice work here. 

But I'm not sure you should use "return". Just to be safe. 

Only use return when sending errors or when sending final succes response.